ibm/resource_ibm_kms_key.go

```diff
@@ -185,13 +192,23 @@ func resourceIBMKmsKeyCreate(d *schema.ResourceData, meta interface{}) error {
 	kpAPI.Config.InstanceID = instanceID
 	name := d.Get("key_name").(string)
 	standardKey := d.Get("standard_key").(bool)
-
+	expiration_string := d.Get("expiration_date").(string)
+	var expiration_time time.Time
+	expiration := &expiration_time
+	if expiration_string != "Default" {
```

Comment:
So all these logic can be moved  loop d.GetOk("XXXXX");ok {
}

Reply:
Sure that would be cleaner.